_inc/lib/core-api/wpcom-endpoints/publicize-services.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+/**
+ * Publicize: List Publicize Services
+ *
+ * [
+ *   { # Service Object. See schema for more detail.
+ *     name:  (string) Service slug
+ *     label: (string) Human readable label for the Service
+ *     url:   (string) Connect URL
+ *   },
+ *   ...
+ * ]
+ *
+ * @since 6.8
+ */
+class WPCOM_REST_API_V2_Endpoint_List_Publicize_Services extends WP_REST_Controller {
+	public function __construct() {
+		$this->namespace = 'wpcom/v2';
+		$this->rest_base = 'publicize/services';
+
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	/**
+	 * Called automatically on `rest_api_init()`.
+	 */
+	public function register_routes() {
+		register_rest_route( $this->namespace, '/' . $this->rest_base, array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_items' ),
+				'permission_callback' => array( $this, 'get_items_permission_check' ),
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+	}
+
+	public function get_item_schema() {
+		$schema = array(
+			'$schema' => 'http://json-schema.org/draft-04/schema#',
+			'title' => 'jetpack-publicize-service',
+			'type' => 'object',
+			'properties' => array(
+				'name' => array(
+					'description' => __( 'Alphanumeric identifier for the Publicize Service', 'jetpack' ),
+					'type' => 'string',
+				),
+				'label' => array(
+					'description' => __( 'Human readable label for the Publicize Service', 'jetpack' ),
+					'type' => 'string',
+				),
+				'url' => array(
+					'description' => __( 'The URL used to connect to the Publicize Service', 'jetpack' ),
+					'type' => 'string',
+					'format' => 'uri',
+				),
+			),
+		);
+
+		return $this->add_additional_fields_schema( $schema );
+	}
+
+	/**
+	 * Retrieves available Publicize Services.
+	 *
+	 * @see Publicize::get_available_service_data()
+	 *
+	 * @param WP_REST_Request $request
+	 * @return WP_REST_Response suitable for 1-page collection
+	 */
+	public function get_items( $request ) {
+		global $publicize;
+		/**
+		 * We need this because Publicize::get_available_service_data() uses `Jetpack_Keyring_Service_Helper`
+		 * and `Jetpack_Keyring_Service_Helper` relies on `menu_page_url()`.
+		 *
+		 * We also need add_submenu_page(), as the URLs for connecting each service
+		 * rely on the `sharing` menu subpage being present.
+		 */
+		include_once( ABSPATH . 'wp-admin/includes/plugin.php' );
```

Comment:
I'd love it if we could get rid of any assumptions that this code is running in wp-admin context. Not in this PR, of course.